src/core_plugins/vega/public/vega_view/vega_view.js

```diff
@@ -0,0 +1,283 @@
+      // FIXME: In some cases, Vega may be initialized twice, e.g. after awaiting... TBD
+      if (!this._$container) return;
+
+      url = tmsService.getUrl();
+      limits = tmsService.getTMSOptions();
+    } else {
+      limits = { minZoom: 0, maxZoom: 25 };
+    }
+
+    const validate = (name, value, dflt, min, max) => {
+      if (value === undefined) {
+        value = dflt;
+      } else if (value < min) {
+        this.onWarn(`Resetting ${name} to ${min}`);
+        value = min;
+      } else if (value > max) {
+        this.onWarn(`Resetting ${name} to ${max}`);
+        value = max;
+      }
+      return value;
+    };
+
+    let minZoom = validate('minZoom', mapConfig.minZoom, limits.minZoom, limits.minZoom, limits.maxZoom);
+    let maxZoom = validate('maxZoom', mapConfig.maxZoom, limits.maxZoom, limits.minZoom, limits.maxZoom);
+    if (minZoom > maxZoom) {
+      this.onWarn('minZoom and maxZoom have been swapped');
+      [minZoom, maxZoom] = [maxZoom, minZoom];
+    }
+    const zoom = validate('zoom', mapConfig.zoom, 2, minZoom, maxZoom);
+
+    // let maxBounds = null;
+    // if (mapConfig.maxBounds) {
+    //   const b = mapConfig.maxBounds;
+    //   maxBounds = L.latLngBounds(L.latLng(b[1], b[0]), L.latLng(b[3], b[2]));
+    // }
+
+    const map = L.map(this._$container.get(0), {
+      minZoom,
+      maxZoom,
+      center: [mapConfig.latitude, mapConfig.longitude],
+      zoom,
+      zoomControl: mapConfig.zoomControl,
+      attributionControl: useDefaultMap,
+      // TODO: test and enable
+      // maxBounds
+    });
+
+    if (useDefaultMap) {
+      map.attributionControl.setPrefix('');
+
+      baseLayer = L
+        .tileLayer(url, {
+          minZoom: limits.minZoom,
+          maxZoom: limits.maxZoom,
+          subdomains: limits.subdomains || [],
+          attribution: limits.attribution
+        })
+        .addTo(map);
+    }
+
+    const vegaLayer = L
+      .vega(this._parser.spec, {
+        vega,
+        bindingsContainer: this._$controls.get(0),
+        delayRepaint: mapConfig.delayRepaint,
+        viewConfig: this._vegaViewConfig,
+        onWarning: this.onWarn.bind(this),
+        onError: this.onError.bind(this),
+      })
+      .addTo(map);
+
+    VegaView.setDebugValues(vegaLayer._view, vegaLayer._spec);
+
+    this._addDestroyHandler(() => {
+      map.removeLayer(vegaLayer);
+      if (baseLayer) map.removeLayer(baseLayer);
+      map.remove();
+    });
+  }
+
+  /**
+   * Set global debug variable to simplify vega debugging in console. Show info message first time
+   */
+  static setDebugValues(view, spec, vlspec) {
+    if (window) {
+      if (window.VEGA_DEBUG === undefined && console) {
+        console.log('%cWelcome to Kibana Vega Plugin!', 'font-size: 16px; font-weight: bold;');
+        console.log('You can access the Vega view with VEGA_DEBUG. Learn more at https://vega.github.io/vega/docs/api/debugging/.');
```

Comment:
I don't think this has a place in Kibana, so you'll have to find a thumbs up for from somebody else :) for these debug logs to go in.

I understand that this makes debugging easy, but right now the Kibana-app is not a developer platform (except maybe the console-app) nor a practice sandbox for the Vega-language.

Reply:
It would make the life of whoever develops Vega vis harder. I think we should provide it, just like we provide the dev console app. And this is completely hidden from the UI unless a dev needs it. If the user is advanced enough to go into the browser's debug console, why not help them with this?

Reply:
CC: @spalger what do you think? Would having a hidden way to help developers debug Vega a big no no for Kibana? This feature is the same as the official Vega editor, and you can only see it if you open the browser's dev console - not something you would do if you are not a dev, and something you are usually ok with if you are a dev.  Of course I could also simply hide the `console.log`, allowing users who know about it to use it (the `window.VEGA_DEBUG` will still be set, and it will be documented), but not giving a hint about it.